### PR TITLE
Documentation improvements for generic types

### PR DESCRIPTION
**Pull request by vshjxyz**

### CODE REVIEW

File: doc/input-streams/input-streams.md [hunk 1]
```diff
@@ -4,13 +4,13 @@ An "input stream" in mutator-io is a class that can create Observables from a sp
 ## Interface
 ```typescript
 interface InputStream<T> {
-  create(): Observable<T>
+  create(): Observable<T> | Promise<T> | T
 }
 ```
 
-Nothing too restrictive. The only requirement is a `create` method that should return an observable.
+Nothing too restrictive. The only requirement is a `create` method that should return the same type provided during the definition of the class (generic `T`).
```
Comment (amidabucu) [inline doc/input-streams/input-streams.md:11]: `The only requirement is a `create` method that should return`, the word `that` is extraneous or should move to after `is`.

Comment (vshjxyz) [reply doc/input-streams/input-streams.md:11]: Sorry I don't get the suggestion, this to me looks like a proper sentence 🤔 
>The only requirement is a `create` method that should return the same type provided during the definition of the class (generic `T`).

Comment (amidabucu) [reply doc/input-streams/input-streams.md:11]: Oh, ok, sorry. My early in the morning brain misparsed that sentence. 😩

**Review by amidabucu — APPROVED** [review]
LGTM

**Pull request merged**